hier_config/base.py

```diff
@@ -119,9 +119,7 @@ def all_children(self) -> Iterator[HConfigChild]:
             yield child
             yield from child.all_children()
 
-    def get_child_deep(
-        self, match_rules: tuple[MatchRule, ...]
-    ) -> Optional[HConfigChild]:
+    def get_child_deep(self, match_rules: tuple[MatchRule, ...]) -> HConfigChild | None:
```

Comment:
I thought Python 3.9 didn't support the `| None` syntactic sugar. https://peps.python.org/pep-0604/ states it was implemented in Python 3.10.